### PR TITLE
MultiQC docs: add modules index page

### PR DESCRIPTION
**Pull request by vladsavelyev**
Use a modules index page built here https://github.com/MultiQC/MultiQC/pull/2903 instead of a sidebar section.

### CODE REVIEW

**Comment by ewels** [comment]
Great work, this looks much better! 👏🏻 

Main downside I see in the deployment preview is that the module pages no longer show the sidebar, eg. [here](https://deploy-preview-246--seqera-docs.netlify.app/multiqc/modules/afterqc). This is a bit of a problem as you suddenly can't navigate elsewhere in the docs.
I think / hope that we can solve this by adding `displayed_sidebar: multiqcSidebar` to the markdown frontmatter ([docs](https://docusaurus.io/docs/sidebar/multiple-sidebars)).

Minor, not blocking: It'd be great to have a filter for the table like in the old page 🤔  Can look into that in the future though.

**Comment by vladsavelyev** [comment]
I think this solves all the problems: https://github.com/MultiQC/MultiQC/pull/2904

Adds a reasonable index (also the "development/index" page is not so uninformative), and the sidebar main items are now clickable.

**Comment by justinegeffen** [comment]
Thanks for this PR! Is it possible to turn the expand arrows in a different direction? They face inward ">" and are large while the other doc sets use small ">"s.

**Comment by vladsavelyev** [comment]
We can tweak CSS for sure, but that's just the default Docusaurus style, and there is a benefit in using the default, as the website immediately looks familiar to those who have seen other Docusaurus backed docs :)

File: multiqc_docs/sidebar.js [hunk 1]
```diff
@@ -50,8 +59,8 @@ export default {
       },
       {
         "type": "category",
-        "label": "Supported tools",
-        "collapsed": true,
+        "label": "Modules",
```
Comment (ewels) [inline multiqc_docs/sidebar.js:62]: I've been trying to get away from the term "Modules" as it means so many different things. With the unified search it'll also come up for nextflow modules, nf-core modules, etc etc.

Is there a reason to move away from "Supported Tools"?

Comment (vladsavelyev) [reply multiqc_docs/sidebar.js:62]: We call them "Modules" elsewhere in the docs and the codebase, e.g. "Writing new modules" in the development section.
Screenshot 2024-10-05 at 12 38 55
So I'd say if we want to move away, we would be moving away from "Modules" :) Happy to give it a shot at least across the modules. E.g. "Writing new modules" -> "Adding tools", etc.

**Comment by justinegeffen** [comment]
> We can tweak CSS for sure, but that's just the default Docusaurus style, and there is a benefit in using the default, as the website immediately looks familiar to those who have seen other Docusaurus backed docs :)

I think we've tweaked the style on our side already - hence the difference. I'm happy with either option but it should be consistent.

File: multiqc_docs/sidebar.js (hunk 1 above)
Comment (ewels) [reply multiqc_docs/sidebar.js:62]: Yeah, maybe you're right and we should move away entirely.

I'd at least like to leave this page as "Supported Tools" anyway. The folks looking for these pages are typically end users and not developers. I think "Supported Tools" is more user friendly and intuitive.

**Comment by ewels** [comment]
There aren't any changes to the styling here, I guess the arrows are reverting because of the addition of `"link": { "type": "generated-index", },` in the config.

@vladsavelyev why was this needed? I think I'll need to go into the theme customisation and add support for this explicitly I guess.

**Comment by ewels** [comment]
And before going much further - whilst I agree this is an improvement I still don't like it as much as the old MultiQC equivalent page.

I would prefer to hand-code a table view that looks like the one we had before, and remove the tool names from the side nav.

We should be able to still have functionality like the filter box etc. We can write custom React components in the Docusaurus theme as needed, then build the HTML / MD in the MultiQC repo accordingly. I can take a look at this if needed, though might be post-summit now.

**Comment by vladsavelyev** [comment]
I added a custom component to display a modules table. It's autogenerated with `scripts/make_module_docs.py` in https://github.com/MultiQC/MultiQC/pull/2903

However, I didn't tweak styles to make it look organic. I need help to improve the design, probably use some website classes and brand colors to match the table and the input field the rest of the pages.

**Comment by vladsavelyev** [comment]
Okay, added better styles and behaviour:

Screenshot 2024-11-07 at 14 04 38

Let me know if it looks good and feel free to push more into the branch @ewels!

**Review by ewels — APPROVED** [review]
Perfect 👌🏻 

Thank you @vladsavelyev !

**Pull request merged**